homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,41 @@
+#include "compression.hpp"
+
+compressedGrayscaleImage compressGrayscale(const grayscaleImage& bitmap) {
+    compressedGrayscaleImage compressed;
+    compressed.reserve(width * height);
+    for (const auto& line : bitmap) {
+        uint8_t counter = 1;
+        for (auto it = line.begin(); it != line.end(); it++) {
+            if (*it != *std::next(it) || it == line.end() - 1) {
+                compressed.push_back(std::make_pair(*it, counter));
+                counter = 1;
+            } else {
+                counter++;
+            }
+        }
+    }
+    return compressed;
+}
+grayscaleImage decompressGrayscale(const compressedGrayscaleImage& compressed) {
+    grayscaleImage decompressed;
+    size_t i = 0;
+    for (auto it : compressed) {
+        for (size_t j = 0; j < it.second; ++j, ++i) {
+            decompressed[0][i] = it.first;
```

Comment:
I don't quite understand, you only fill values to first row of decompressed image? As far as I know `grayscaleImage` consists of `height` (32) rows. What if `i` values will be greater than `width`?